### DISCUSSION TITLE
package is package, software is software

### DISCUSSION THREAD

**Discussion by avelino** [prest/prest#190] [API Server] [answered]
The **pREST** has grown very much and we can diversify the usage, it is not just a monolithic software it is possible to use it as a framework (modules https://github.com/nuveo/prest/blob/master/CREATING_MODULES.md) as well.

Today we have an architectural error using the **pREST** as framework (module) because we have the `vendor` folder.

With the point raised above, I propose we migrate the **pREST** to a new organization (https://github.com/prest) and divide it into package, the `prest/prest` as software and all its modules in isolated repositories (with isolated tests)

I moved all the history commit of the `adapter` package to this repository (as example): https://github.com/prest/adapters

- Is that a good idea?
- Does that solve our problems?

**Comment by marioidival** [comment]
Ruby On Rails did this before and they were successful. Let's go!

**Comment by marioidival** [comment, marked as answer]
- [x] prest/prest
- [x] https://github.com/prest/adapters/pull/1
- [x] https://github.com/prest/cmd/pull/1
- [x] https://github.com/prest/config/pull/1
- [x] https://github.com/prest/middlewares/pull/1
- [x] https://github.com/prest/controllers/pull/1

**Comment by avelino** [comment]
Congratulations team, so it is easy to pREST update for framework (as we are using within the @nuveo)
